src/components/ec-date-range-filter/ec-date-range-filter.vue

```diff
@@ -50,101 +50,102 @@
-      type: String,
-      required: false,
-      default: '',
-    },
-    dateRangeErrorMessage: {
-      type: String,
-      required: false,
-      default: '',
-    },
-    popoverOptions: {
-      type: Object,
-    },
-    modelValue: {
-      type: Object,
-    },
+const props = defineProps({
+  label: {
+    type: String,
+    required: true,
+    default: '',
+  },
+  fromLabelText: {
+    type: String,
+    required: false,
+    default: 'From',
+  },
+  toLabelText: {
+    type: String,
+    required: false,
+    default: 'To',
+  },
+  clearText: {
+    type: String,
+    required: false,
+    default: 'Clear dates',
+  },
+  fromErrorMessage: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  toErrorMessage: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  dateRangeErrorMessage: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  popoverOptions: {
+    type: Object,
+  },
+  modelValue: {
+    type: Object,
+  },
+});
+
+const emit = defineEmits(['update:modelValue', 'change', 'blur']);
+
+// update selected dates
+function update(newValue) {
+  emit('update:modelValue', newValue);
+  emit('change', newValue);
+}
+const fromValueDate = computed({
+  get() {
+    return props.modelValue?.from;
   },
-  emits: ['update:modelValue', 'change', 'blur'],
-  computed: {
-    fromValueDate: {
-      get() {
-        return this.modelValue?.from;
-      },
-      set(value) {
-        this.update({ from: value, to: this.toValueDate });
-      },
-    },
-    toValueDate: {
-      get() {
-        return this.modelValue?.to;
-      },
-      set(value) {
-        this.update({ from: this.fromValueDate, to: value });
-      },
-    },
-    numberOfSelectedFilters() {
-      let datesSelected = 0;
-      if (this.fromValueDate && this.toValueDate) {
-        datesSelected = 2;
-      } else if (this.fromValueDate || this.toValueDate) {
-        datesSelected = 1;
-      }
-      return datesSelected;
-    },
-    isDisabled() {
-      return !this.numberOfSelectedFilters > 0;
-    },
+  set(value) {
+    update({ from: value, to: toValueDate.value });
   },
-  methods: {
-    update(newValue) {
-      this.$emit('update:modelValue', newValue);
-      this.$emit('change', newValue);
-    },
-    clear() {
-      this.update(null);
-    },
-    onBlur() {
-      this.$emit('blur', { from: this.fromValueDate, to: this.toValueDate });
-    },
+});
+const toValueDate = computed({
+  get() {
+    return props.modelValue?.to;
   },
-};
+  set(value) {
+    update({ from: fromValueDate.value, to: value });
+  },
+});
+
+// disable clear button
+const numberOfSelectedFilters = computed(() => {
+  let datesSelected = 0;
+  if (fromValueDate.value && toValueDate.value) {
+    datesSelected = 2;
+  } else if (fromValueDate.value || toValueDate.value) {
+    datesSelected = 1;
+  }
+  return datesSelected;
+});
+const isDisabled = computed(() => !numberOfSelectedFilters.value > 0);
```

Comment:
```suggestion
const isButtonDisabled = computed(() => numberOfSelectedFilters.value <= 0);
```
It is better and more natural for our mind to avoid and understand easier a positive not a negative conditions. Name of variable should explain itself. This makes reading of the code more fluent and doesn't force reader to jump from script to template and vice versa to understand what isDisabled for.